### DISCUSSION TITLE
Add no-empty-function from ESLint

### DISCUSSION THREAD

**Discussion by floisloading** [rome/tools#4345] [Suggestions]
The Angular CLI generates components like this:
 ```
export class FreeTrialFormBlaComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }

}
```

Most of the time (for view components), these two methods will stay empty. ESLint has a rule to prevent empty methods: https://eslint.org/docs/latest/rules/no-empty-function

Can this be implemented in Rome?

**Comment by jpike88** [comment]
was just about to add this one! very useful.

**Comment by Conaclos** [comment]
TypeScript ESLInt [extends the rule](https://typescript-eslint.io/rules/no-empty-function/).

I am wondering: why ESLint separates [no-empty-function](https://eslint.org/docs/latest/rules/no-empty-function) from [no-empty](https://eslint.org/docs/latest/rules/no-empty) and [no-empty-static-block](https://eslint.org/docs/latest/rules/no-empty-static-block)?

This could be a single rule. This could also integrate [no-empty-interface](https://typescript-eslint.io/rules/no-empty-interface/).

**Reply by jpike88** [reply]
I agree, empty anything is kind of useless, a single rule against all of that is good to me
